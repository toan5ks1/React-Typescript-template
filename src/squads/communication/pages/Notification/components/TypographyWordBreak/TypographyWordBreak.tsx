import { Theme } from "@mui/material";
import TypographyBase, { TypographyBaseProps } from "src/components/Typographys/TypographyBase";

export interface TypographyWordBreakProps extends TypographyBaseProps {
    isTextInAccordion?: boolean;
}

const sx = {
    accordion: (theme: Theme) => ({
        /**
         * @param theme.spacing(3): width of IconExpand Accordion
         * @param theme.spacing(1): marginLeft of IconExpandAccordion
         */
        width: `calc(100% - ${theme.spacing(3)} - ${theme.spacing(1)})`,
    }),
    wordBreak: () => ({
        wordBreak: "break-word",
    }),
};

const TypographyWordBreak = (props: TypographyWordBreakProps) => {
    const { isTextInAccordion, sx: sxTypographyBase = [sx.wordBreak], ...rest } = props;

    return (
        <TypographyBase
            sx={[isTextInAccordion ? sx.accordion : {}, ...(Array.isArray(sx) ? sx : [sx])]}
            {...rest}
        />
    );
};

export default TypographyWordBreak;
