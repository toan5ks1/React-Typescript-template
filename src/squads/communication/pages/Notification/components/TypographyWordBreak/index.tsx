export { default } from "./TypographyWordBreak";
export * from "./TypographyWordBreak";
