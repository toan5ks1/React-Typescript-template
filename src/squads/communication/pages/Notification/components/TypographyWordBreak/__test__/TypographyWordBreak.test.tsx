import TypographyWordBreak from "../TypographyWordBreak";

import { render } from "@testing-library/react";

describe("<TypographyWordBreak />", () => {
    it("should match snapshot", () => {
        const wrapper = render(<TypographyWordBreak />);
        expect(wrapper.container).toMatchSnapshot();
    });

    it("should match snapshot when typography's text is in accordion", () => {
        const wrapper = render(<TypographyWordBreak isTextInAccordion={true} />);
        expect(wrapper.container).toMatchSnapshot();
    });
});
