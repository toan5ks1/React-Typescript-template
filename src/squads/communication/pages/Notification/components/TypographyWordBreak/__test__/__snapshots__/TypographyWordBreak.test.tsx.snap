// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TypographyWordBreak /> should match snapshot 1`] = `
.emotion-0 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
}

.emotion-0 accordion {
  width: calc(100% - 24px - 8px);
}

.emotion-0 wordBreak {
  word-break: break-word;
}

<div>
  <p
    class="MuiTypography-root MuiTypography-body1 emotion-0"
  />
</div>
`;

exports[`<TypographyWordBreak /> should match snapshot when typography's text is in accordion 1`] = `
.emotion-0 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  width: calc(100% - 24px - 8px);
}

.emotion-0 accordion {
  width: calc(100% - 24px - 8px);
}

.emotion-0 wordBreak {
  word-break: break-word;
}

<div>
  <p
    class="MuiTypography-root MuiTypography-body1 emotion-0"
  />
</div>
`;
